app/uk/gov/hmrc/gform/models/helpers/Fields.scala

```diff
@@ -109,6 +109,7 @@ object Fields {
               FieldError(fieldValue, fileName, errors))
         }
       case InformationMessage(_, _) => None
+      case HmrcTaxPeriod(_, _, _)   => evalChoice(fieldValue, gformErrors)(dataGetter)
```

Comment:
This is suspicious. Is `HmrcTaxPeriod` component really so similar to `Choice` component that we can use methods tailored for the latter?